packages/demos/stories/DragDrop.tsx

```diff
@@ -0,0 +1,135 @@
+/**
+ * © 2020 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayList from '@clayui/list';
+import {XYCoord} from 'dnd-core';
+import update from 'immutability-helper';
+import React, {useState, useCallback, useRef} from 'react';
+import {useDrag, useDrop, DropTargetMonitor} from 'react-dnd';
+
+export interface IDraggableListItemProps {
+	/**
+	 * The text displayed in the List Item
+	 */
+	text: string;
+	/**
+	 * Index of the List Item in order to keep track of it's positioning within the array
+	 */
+	index: number;
+	/**
+	 * Function that handles the moving of List Items
+	 */
+	moveListItem: (dragIndex: number, hoverIndex: number) => void;
+}
+
+interface IDragItem {
+	index: number;
+	type: string;
+}
+
+const DraggableListItem: React.FC<IDraggableListItemProps> = ({
+	index,
+	moveListItem,
+	text,
+}) => {
+	const ref = useRef<HTMLDivElement>(null);
+
+	const [, drop] = useDrop({
+		accept: 'listItem',
+		hover(item: IDragItem, monitor: DropTargetMonitor) {
+			if (!ref.current) {
+				return;
+			}
+
+			const dragIndex = item.index;
+
+			const hoverIndex = index;
+
+			if (dragIndex === hoverIndex) {
+				return;
+			}
+
+			const hoverBoundingRect = ref.current!.getBoundingClientRect();
+
+			const verticalMiddle =
+				(hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
+
+			const mousePosition = monitor.getClientOffset();
+
+			const pixelsToTop =
+				(mousePosition as XYCoord).y - hoverBoundingRect.top;
+
+			if (dragIndex < hoverIndex && pixelsToTop < verticalMiddle) {
+				return;
+			}
+
+			if (dragIndex > hoverIndex && pixelsToTop > verticalMiddle) {
+				return;
+			}
+
+			moveListItem(dragIndex, hoverIndex);
+
+			item.index = hoverIndex;
+		},
+	});
+
+	const [{isDragging}, drag] = useDrag({
+		collect: (monitor: any) => ({
+			isDragging: monitor.isDragging(),
+		}),
+		item: {index, type: 'listItem'},
+	});
+
+	const opacity = isDragging ? 0 : 1;
+
+	drag(drop(ref));
+
+	return (
+		<div ref={ref} style={{opacity}}>
+			<ClayList.Item>{text}</ClayList.Item>
```

Comment:
Instead of just using this, lets make it a full list like we see in https://storybook.clayui.com/?path=/story/components-claylist--simple. no need for headers though.